src/metabase/models/data_permissions.clj

```diff
@@ -556,6 +556,9 @@
 
 (defn- assert-valid-permission
   [{:keys [perm_type perm_value] :as permission}]
+  (when (= perm_value :legacy-no-self-service)
+    (throw (ex-info (tru "Permission value {0} is deprecated and can not be set." :legacy-no-self-service)
```

Comment:
Is there any chance we'd want to enforce this on the frontend but not the backend, to allow us to write e2e or other tests that set `legacy-no-self-service` and make assertions about behavior?

Reply:
That's a good idea. Since we'll be communicating that this is deprecated there shouldn't be a risk of customers setting it via the API (and if they do, it's at their own risk)